src/Repl/Repl.tsx

```diff
@@ -311,15 +311,15 @@ const Repl = (props) => {
       let pc = props.programCode;
       // only run this when we have program code (in testing this is not defined)
       if(pc !== undefined) {
-        let pCheck = pc.replaceAll(/(?:--[^\n]*)/g,'').trim();
+        let pCheck = removeComments(pc)
         if(pCheck === "") {
           // empty program, substitute a default program so we can run just expressions
           pc = "game Expressions";
         }
       }
 
-      // extract game name, and in the case of an undefined program (when testing), supply a default definition to work with
-      let gameName = extractGameNameFromProgram(pc !== undefined ? pc : "game DefaultTestingDefinition");
+      // extract game name, and in the case of an undefined program (when testing), supply a default name to work with
+      let gameName = pc !== undefined ? extractGameNameFromProgram(pc) : "DefaultTestingDefinition";
```

Comment:
Realized there was no point in doing extra work to extract the name from a dummy "game Default..." string, especially if we don't use it anywhere besides here. So the extraction is only performed on legitimate programs.